File: python/6_kyu/find_the_odd_int.py
# https://www.codewars.com/kata/54da5a58ea159efa38000836

def find_it(seq):
    values, hgs, intg = {}, 0, None
    for x in seq:
        values[x] = 1 if x not in values else values[x] + 1
    for number in values:
        if values[number] > hgs and (values[number] % 2 >= 1):
            intg, hgs = number, values[number]
    return intg

File: c#/6_kyu/FindTheOddInt.cs
namespace Solution
{
  class Kata
    {
    public static int find_it(int[] seq) 
      {
        foreach (var n in seq)
          if (CountOccurences(seq, n) % 2 != 0)
            return n;
        return 0;
      }
    static int CountOccurences(int[] array, int element) 
      {
        var count = 0;
        foreach (var n in array)
          if (n == element)
            count++;
        return count;
      }
    }
}

// https://www.codewars.com/kata/54da5a58ea159efa38000836/train/csharp